### DISCUSSION TITLE
Redirect user to respective panel.

### DISCUSSION THREAD

**Discussion by yuvrajtimalsina** [filamentphp/filament#12238] [Help]
### Package

Panel builder

### Package Version

v3.2.57

### How can we help you?

I have User model with column named role and I have created multiple Panels for each role.
I also created a custom middleware 'CheckRole' to redirect user based on role in request.

But since I plan on using single middleware for the cause, I need to pass role in my middleware and I am unable to achieve it.

```->authMiddleware([
                Authenticate::class,
                CheckRole::class,
            ]);
```
How do i pass parameter to CheckRole::class in AdminPanelProvider filamentphp?

```
  public function handle(Request $request, Closure $next, Role $requiredRole): Response
    {
        $userRole = $request->user()->role;

        if ($userRole !== $requiredRole) {
            $selfPanelRoute = Filament::getPanel($userRole->value)->getId();

            return redirect($selfPanelRoute);
        }

        return $next($request);
    }
```
I have multiple panel and I want to implement same middleware for both.

**Comment by albetnov** [comment]
You can add a login to each of the panels. Alternatively, if you want a single login page you can create a custom login action that extends the base login action. You can adjust the redirection there. 

For checking the role, you can remove the middleware and instead implement `FilamentUser` to your model and adjust the corresponding access to the panel id for each user based on their role.

See https://filamentphp.com/docs/3.x/panels/users#authorizing-access-to-the-panel, https://filamentphp.com/docs/3.x/panels/users#customizing-the-authentication-features

**Reply by yuvrajtimalsina** [reply]
Is there a way, I could leverage middleware for the cause?

**Reply by taher-k5** [reply]
https://v2.filamentphp.com/tricks/customize-redirect-after-admin-login
use this trick

**Comment by taher-k5** [comment]
Use this to redirect user to their respective dashboard
https://v2.filamentphp.com/tricks/customize-redirect-after-admin-login
